mlprimitives/timeseries.py

```diff
@@ -0,0 +1,67 @@
+import pandas as pd
+import numpy as np
+
+
+def rolling_window_sequences(X, window_size, value_column, time_column):
+    """
+        Function that takes in a pandas.DataFrame and a window_size then creates
+            output arrays that correspond to a timeseries sequence with window_size overlap.
+            The output arrays can be fed into a timeseries forecasting model.
+            Assumes the input is timeseries sorted.
+        Args:
+            X (pandas.DataFrame): a pandas dataframe which has 'timestamp'
+                and 'value' columns, and is sorted based on timestamp. 
+                The timestamp column is in UNIX format (in seconds).
+            window_size (int): number of values that overlap to create the sequence.
+            value_column (string): name of column that has the value field.
+            time_column (string): name of column that has the time field.
+        Returns:
+            (numpy.ndarray): contains the time series sequenced data with each 
+                entry having window_size rows.
+            (numpy.ndarray): acts as the label for the forecasting problem with 
+                each entry having window_size rows.
+            (numpy.ndarray): the corresponding timestamps series.
+    """
+    output_X = []
+    Y = []
+    time = []
+    for i in range(len(X) - window_size):
+        # reshape into a vector to fit into a neural network model (vectorize it)
+        output_X.append(X[i: i + window_size][value_column].values.copy().reshape([-1, 1])) 
```

Comment:
Is the `copy()` bit necessary? I think we can safely skip it and relax a bit the memory overhead.

Reply:
yep I think it's safe to do so, since we aren't altering the dataframe elsewhere.